adaptive_ai/tests/final_prompt_code_generation.py

```
@@ -0,0 +1,236 @@
+# Select only Prompt and Difficulty columns
+filtered_ds = ds["train"].select_columns(["Question", "Difficulty"])
+
+# Count difficulty levels
+difficulty_counts = Counter(filtered_ds["Difficulty"])
+print("\nDifficulty level distribution:")
+for difficulty, count in difficulty_counts.items():
+    print(f"{difficulty}: {count} examples")
+
+print("\nTotal examples by difficulty:")
+for difficulty in ["Easy", "Hard"]:
+    count = len(filtered_ds.filter(lambda x: x["Difficulty"] == difficulty))
```

Ruff (0.11.9)

Function definition does not bind loop variable `difficulty`

(B023)